### PR TITLE
unhandled promise errors fixed

### PR DESCRIPTION
**Pull request by lospennikov-nbl**
Fix Issue #28 from electricimp/Promise

### CODE REVIEW

**Comment by ppetrosh** [comment]
Does it have handle the serial case:

```
Promise.serial([
        Promise(function(resolve, reject) {_}),
        Promise(function(resolve, reject) {resolve("1")})
    ]).then(null, null)
```
?

**Comment by ppetrosh** [comment]
Also does the fix address the case when you call `then`, with null handlers provided:

`
Promise(function(resolve, reject) {_})
.then(null, null);
`
?

**Comment by ppetrosh** [comment]
The following code:

```
Promise.serial([
        Promise(function(resolve, reject) {_}),
        Promise(function(resolve, reject) {server.log("1");resolve("1")}),
        Promise(function(resolve, reject) {server.log("2"); resolve("2")}),
        Promise(function(resolve, reject) {server.log("3"); resolve("3")})
    ]).then(null, null)
```

produces the following output:

```
2017-03-17 22:36:31+0000 [Server] Agent restarted: reload.
2017-03-17 22:36:32+0000 [Server] Device connected
2017-03-17 22:36:32+0000 [Device] 1
2017-03-17 22:36:32+0000 [Device] 2
2017-03-17 22:36:32+0000 [Device] 3
2017-03-17 22:36:32+0000 [Device] Unhandled promise rejection: the index '_' does not exist (line 340)
2017-03-17 22:36:32+0000 [Device] Unhandled promise rejection: the index '_' does not exist (line 340)
```

which I guess is incorrect as it should be just one message printed.

**Comment by lospennikov-nbl** [comment]
Yes, fixed

File: promise.class.nut [hunk 1]
```diff
@@ -42,6 +48,9 @@ class Promise {
     function _callHandlers() {
         if (this.STATE_PENDING != this._state) {
             imp.wakeup(0, function() {
+                if ((this._handlers.len() == 0) && (this._last) && (this.STATE_REJECTED == this._state)) {
```
Comment (snork-nbl) [inline promise.class.nut]:
```
if (this._last 
    && this._handlers.len() == 0
    && this.STATE_REJECTED  == this._state) {
```

**Comment by snork-nbl** [comment]
Please update the LICENSE file and the headers and I'll approve it. Thanks!

**Pull request merged**